### PR TITLE
Add orogen debug configuration provider

### LINKED ISSUE #10: contribute debugging configuration to allow easy creation of debugging launchers for orogen components
In case the user has more specific configurations/setup he/she would be able to create custom debugging launchers in launch.json and still get the extension to take care of boilerplate stuff

### LINKED ISSUE #14: Re-discuss the workflow
I generally don't like doing things differently than the platform we're running on. VSCode has UI paths to build and debug, we should use them and not create parallel ones. The build and debug buttons in the status bar look like a case of "I like my way better than the VS people's way" IMO.

### PR DESCRIPTION
**Pull request by g-arjones**
I am currently using this for my orogen-related activities. Personally, I think it works much better than the current scheme that relies on rock-run and it also allows me to debug more than one component simultaneously. If you think this is acceptable, I can go all the way and remove all the alternative mechanisms (build, launch, package selection, statusbar) and you can rebase #38 and do your work on a (much) cleaner codebase.

This closes #10 and closes #14

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,52 @@
         "roby"
     ],
     "contributes": {
+        "debuggers": [
```
Comment (doudou) [inline package.json]: I didn't realize one could "serve" a custom debugger by a configuration provider. Really neat.

**Comment by doudou** [comment]
Since the work is done, I don't have a problem with merging it, even if I do plan to remove it _in fine_.

Do go all the way, it will clean things up tremendously.

**Comment by g-arjones** [comment]
> it will clean things up tremendously.

Do you want to keep "package selection" and/or the statusbar?

**Comment by doudou** [comment]
> Do you want to keep "package selection" and/or the statusbar?

The underlying question I guess is, what would they be still useful for ?

**Comment by doudou** [comment]
> Since the work is done, I don't have a problem with merging it, even if I do plan to remove it in fine.

Actually, not entirely true. I might want to reuse this.... I like the attach/launch dichotomy in the launch configurations.

The idea would be:
- if "launch", the extension asks the command line to Syskit, and Syskit assumes vscode will start the task
- if "attach", then the extension either gets a PID from Syskit, or a gdbserver port and Syskit starts the deployment accordingly

**Comment by g-arjones** [comment]
> The underlying question I guess is, what would they be still useful for ?

Package selection just to avoid keep asking which package will be affected by a command. And the statusbar just to show which package is selected. Personally, I don't it is worth it and my idea was to remove everything...

> The idea would be:

I like that

**Comment by doudou** [comment]
> Personally, I don't it is worth it and my idea was to remove everything...

Then follow your idea. I currently don't "see" what would be the effect, so let's follow your guts.

**Comment by g-arjones** [comment]
That is it.. Unit tests for the debugging configuration providers are missing (currently only the base class is unit tested).

**Pull request merged**